internal/cli/atlas/kubernetes/config/generate.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"context"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/kubernetes/operator"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type GenerateOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	clusterName     []string
+	orgID           string
+	includeSecrets  bool
+	targetNamespace string
+	store           store.AtlasOperatorGenericStore
+	credsStore      store.CredentialsGetter
+}
+
+func (opts *GenerateOpts) initStores(ctx context.Context) func() error {
+	return func() error {
+		var err error
+
+		profile := config.Default()
+		opts.store, err = store.New(store.AuthenticatedPreset(profile), store.WithContext(ctx))
+		if err != nil {
+			return err
+		}
+		opts.credsStore = profile
+
+		return nil
+	}
+}
+
+func (opts *GenerateOpts) Run() error {
+	result, err := operator.NewConfigExporter(opts.store, opts.credsStore, opts.ProjectID, opts.orgID).
+		WithClustersNames(opts.clusterName).
+		WithTargetNamespace(opts.targetNamespace).
+		WithSecretsData(opts.includeSecrets).
+		Run()
+	if err != nil {
+		return err
+	}
+	return opts.Print(result)
+}
+
+// GenerateBuilder builds a cobra.Command that can run as:
+// atlas kubernetes config generate --projectId=projectId --clusterName="cluster-1,cluster-2...cluster-N" --includeSecrets --targetNamespace=my-namespace.
+func GenerateBuilder() *cobra.Command {
```

Comment:
you should be able to define an e2e test to test your command. see https://github.com/mongodb/mongodb-atlas-cli/tree/master/test/e2e/atlas for examples